fnet/network_test.go

```diff
@@ -169,6 +169,37 @@ func TestResolveIpV6(t *testing.T) {
 	}
 }
 
+func TestJoinHostAndPort(t *testing.T) {
+	var tests = []struct {
+		inputPort string
+		addr      *net.TCPAddr
+		expected  string
+	}{
+		{":8080", &net.TCPAddr{
+			IP:   []byte{192, 168, 2, 3},
+			Port: 8081,
+		}, "192.168.2.3:8081"},
+		{":8081", &net.TCPAddr{
+			IP:   []byte{192, 168, 30, 14},
+			Port: 8080,
+		}, "192.168.30.14:8080"},
+		{":8080",
```

Comment:
missing the case for say "8080" (no : ) and nil address
(which will fail to do localhost:8080 )

Reply:
remove the nil case but do keep 8080 :8080 and [::1]:8080 as inputPort tests